### PR TITLE
feat(sources): Add filter functionality

### LINKED ISSUE #649: Sources filter functionality
**Describe the bug**
There are some issues with filtering for the new sources page.

1. There is no results row when a filter is applied. As a user, I'm expecting to clear filters similarly to the details pages. Currently, the only way to clear a filter is to delete the previous filter string and press enter.

2. The filter does not accept partial strings. This could be an issue for the back-end API to solve. (In that case, please file an issue with koku.) However, I could not match "hccm", only exact matches (e.g., "hccm-vmware") are found.

**To Reproduce**
1. Go to 'Overview'
2. Click on 'Add Sources'
3. Enter a filter such as 'hccm'

**Expected behavior**
As a user, I should be able to clear individual filters. I should also be able to add partial filters.

**Screenshots**

Partial string:
Screen Shot 2019-04-01 at 9 47 59 AM

Full String:
Screen Shot 2019-04-01 at 9 48 14 AM

**Additional context**
See https://docs.google.com/document/d/1_t1dV00l9j529vSWduzAkqtnGlLtlFE7e8zidylkGKo/edit?ts=5ca4eef3

### PR DESCRIPTION
**Pull request by boaz0**
closes #649

### CODE REVIEW

File: src/pages/sourceSettings/filterToolbar.tsx [hunk 1]
```diff
@@ -23,7 +23,7 @@ class FilterToolbar extends React.Component<Props> {
   }
 
   public checkEnter(event: React.KeyboardEvent) {
-    if (event.key === 'Enter') {
+    if (event.key === 'Enter' && this.props.value) {
```
[the remark this answers is not in the document]
Comment (boaz0) [reply src/pages/sourceSettings/filterToolbar.tsx]: Isn't clicking on "clear all filters"  sufficient?

File: src/pages/sourceSettings/sourceSettings.tsx [hunk 2]
```diff
@@ -176,41 +228,55 @@ class SourceSettings extends React.Component<Props, State> {
       <div className={css(styles.sourceSettings)}>
         <Header t={t} />
         <div className={css(styles.content)}>
-          {status === FetchStatus.complete && (rows.length > 0 || filterValue) && (
+          {this.props.dirty && (
```
[the remark this answers is not in the document]
Comment (boaz0) [reply src/pages/sourceSettings/sourceSettings.tsx]: Sorry, I did some checking with it and forgot to remove it.

**Comment by boaz0** [comment]
>    Select 'Type', then enter 'Red'. No partial match.

The backend will return error message if type is not 'AWS' or 'OCP'. Thus, I changed it from text input to selector. Should solve this problem.

>    Select 'Type', then enter 'AWS'. The menu resets to 'Name'.

You're right. I fixed that issue.

>    The current filter is not reflected in the URL.

Not one of the requirements of #649. I will open a new ticket for that, though.

File: src/pages/sourceSettings/filterToolbar.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (dlabrecq) [reply src/pages/sourceSettings/filterToolbar.tsx]: I don't believe that option was available

**Comment by dlabrecq** [comment]
Attempted to run the latest code locally, but encountered the error below.

Screen Shot 2019-05-13 at 10 24 45 AM

**Comment by boaz0** [comment]
It seems like the `translate` function doesn't wrap the component correctly as `translate()(...)` returns a new component that doesn't expects props.

File: src/pages/sourceSettings/filterToolbar.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (boaz0) [reply src/pages/sourceSettings/filterToolbar.tsx]: Maybe it is hard to notice it but you can see here:

[the remark this answers is not in the document]
Comment (dlabrecq) [reply src/pages/sourceSettings/filterToolbar.tsx]: Whatever the issue was, it appears to be working now.

That said, I did find a new localization new issue with the type menu. If you can quickly fix that, I'll approve. 

Screen Shot 2019-05-14 at 9 53 58 AM

**Comment by boaz0** [comment]
Thanks @dlabrecq

**Pull request merged**